### PR TITLE
Android tap list

### PR DESCRIPTION
**Pull request by gleono**
I think I got a basic implementation of the Tap List. We need to test this on an actual device because I was having trouble with Picasso and the Android Simulator.

In the mean time I think this is a good start, from the architecture point of view. So we can start integrating stripe @gwoodard3 :)

### CODE REVIEW

File: androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/activities/BaseActivity.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.hopsquad.hopsquadapp.activities;
+
+import android.support.v7.app.AppCompatActivity;
+
+/**
+ * Created by memo on 15/11/17.
+ */
+
+public class BaseActivity extends AppCompatActivity {
```
Comment (rayolanderos) [inline androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/activities/BaseActivity.java:9]: Was this move just for organization purposes or are you thinking they'll eventually share more things?

File: androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/api/BarTenderModule.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import dagger.Module;
+import dagger.Provides;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+/**
+ * Created by memo on 15/11/17.
+ */
+@Module
+public class BarTenderModule {
```
Comment (rayolanderos) [inline androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/api/BarTenderModule.java:12]: Is the "bartender" the ordering system?

File: androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/api/BeerRepository.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+/**
+ * Created by memo on 15/11/17.
+ */
+
+public class BeerRepository {
+
+    // This is for debugging purposes on the simulator
+    // TODO: switch to actual project on Google Cloud
+    public static final String BASE_URL = "http://10.0.2.2:8080/";
```
Comment (rayolanderos) [inline androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/api/BeerRepository.java:25]: The app has some values ready in the api, should we update this now?

File: androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/viewmodels/TapListViewModel.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        orderList.clear();
+    }
+
+    public void getAndImageFromUrl(String url, ImageView view) {
+
+
+
+    }
+
+//    // based on: https://android-developers.googleblog.com/2010/07/multithreading-for-performance.html
```
Comment (rayolanderos) [inline androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/viewmodels/TapListViewModel.java:77]: wishful hoping for the future?

File: androidapp/HopSquadApp/app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -29,4 +29,17 @@
     <string name="payment_cvv">CVV</string>
     <string name="payment_expiration">Exp</string>
 
+    <string-array name="beer_order_amount">
```
Comment (rayolanderos) [inline androidapp/HopSquadApp/app/src/main/res/values/strings.xml:32]: Found out there may be a TABC restriction about how many beers a single user can order, but this is a business and not technical edit. I think we can leave for now just to show.

**Review by rayolanderos — APPROVED** [review]
Looks good. Just wondering if we should update the base url to the real webapp now. Your call.

File: androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/activities/BaseActivity.java (hunk 1 above)
Comment (gleono) [reply androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/activities/BaseActivity.java:9]: Both, I was having some trouble with fragments and activities versions and I had to change the base Activity class a bunch of times, so instead added this one and only changed it once for all activities.

File: androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/api/BarTenderModule.java (hunk 2 above)
Comment (gleono) [reply androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/api/BarTenderModule.java:12]: It was, but I already deleted it in the other branch I've been working on.

File: androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/api/BeerRepository.java (hunk 3 above)
Comment (gleono) [reply androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/api/BeerRepository.java:25]: I'll add that in the new branch `android-orders`.

File: androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/viewmodels/TapListViewModel.java (hunk 4 above)
Comment (gleono) [reply androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/viewmodels/TapListViewModel.java:77]: I ended up deleting this on the new branch.

File: androidapp/HopSquadApp/app/src/main/res/values/strings.xml (hunk 5 above)
Comment (gleono) [reply androidapp/HopSquadApp/app/src/main/res/values/strings.xml:32]: That is god to know.

**Comment by gleono** [comment]
Yes, but, I'll do it in the other branch which builds on top of this one.

**Pull request merged**